### DISCUSSION TITLE
link_to resource_path

### DISCUSSION THREAD

**Discussion by sageworksstudio** [avo-hq/avo#3392] [Help] [answered]
I have a card partial in my dashboard that I want to add some links to various resources. When I specify the URL as a string it works as expected. But trying to use an Avo rout I get errors. What am I missing?

This works as expected:

`<%= link_to "Orders", "/clients/resources/client_orders" %>`

This gives the error: ActionView::Template::Error (undefined local variable or method `resources_client_orders_path' for an instance of #<Class:0x00007121c4489750>)

`<%= link_to "Orders", resources_client_orders_path %>`

This gives the error: ActionView::Template::Error (undefined method `resources_client_orders_path' for an instance of ActionDispatch::Routing::RoutesProxy)

`<%= link_to "Orders", main_app.resources_client_orders_path %>`

![screenshot-127_0_0_1_3000-2024_11_06-12_07_46](https://github.com/user-attachments/assets/43117883-cd6e-4128-b2bc-b20bbf57927f)

**Comment by Paul-Bob** [comment, marked as answer]
Hey @sageworksstudio 

Because you're in a Rails engine, you must prepend the `avo` engine object to the path

Instead of writing `resources_client_orders_path` you have to write `avo.resources_client_orders_path`.

Use `main_app.some_path` when you want to reference paths from your main app
